src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -327,14 +327,9 @@ private static bool IsHostRelative(string path)
 
         protected override async Task<bool> HandleForbiddenAsync(ChallengeContext context)
```

Comment:
Shouldn't this completely mimic HandleUnauthorizedAsync? e.g. allowing an alternate redirect uri, etc..